### PR TITLE
Guidance around CSS architecture

### LINKED ISSUE #83: Architecting CSS guidance section

### PR DESCRIPTION
**Pull request by craigjbass**
closes #83

### CODE REVIEW

File: core-skills/frontend-web-development/resources/css-architecture.md [hunk 1]
```diff
@@ -1,23 +1,67 @@
...
+
 - http://atomicdesign.bradfrost.com/table-of-contents/
 - https://smacss.com/book/
+- https://maintainablecss.com/chapters/semantics/
+- https://css-tricks.com/when-using-important-is-the-right-choice/
+
+## CSS Frameworks
 
-## SASS
+Once you have blahed
```
Comment (craigjbass) [inline core-skills/frontend-web-development/resources/css-architecture.md]: Not sure what the guidance here should be

File: core-skills/frontend-web-development/resources/css-architecture.md [hunk 2]
```diff
@@ -1,23 +1,67 @@
...
+- https://hacks.mozilla.org/2016/04/you-might-not-need-a-css-framework/
+- https://css-tricks.com/what-are-the-benefits-of-using-a-css-framework/
+
+## CSS Preprocessors
 
 - https://designmodo.com/introduction-sass/
 - https://lincolnloop.com/blog/linting-scss-sass-lint/
 
-## CSS Resets
+## CSS Layout
```
Comment (craigjbass) [inline core-skills/frontend-web-development/resources/css-architecture.md]: Not sure about this section? Is this under grids?

File: core-skills/frontend-web-development/resources/css-architecture.md [hunk 3]
```diff
@@ -1,23 +1,67 @@
...
+### BEM
+
+Block Element Modifier.
 
-- https://maintainablecss.com/chapters/semantics/
-- https://css-tricks.com/when-using-important-is-the-right-choice/
-- https://css-tricks.com/snippets/css/a-guide-to-flexbox/
 - http://getbem.com/
+
+### we also give a mention to...
```
Comment (craigjbass) [inline core-skills/frontend-web-development/resources/css-architecture.md]: Wondering whether we should be a little bit opinionated about BEM and Component based (for beginners). Just so people aren't given a paradox of choice

File: core-skills/frontend-web-development/resources/css-architecture.md [hunk 4]
```diff
@@ -1,23 +1,67 @@
...
+A well-written set of stylesheets balance a number of factors including
+
+* Browser support - does the styling render the same in a wide range of commonly used browsers?
+* Ease of change - can we easily change the styles to meet changing customer needs?
+* Reuse - can we reuse commonly used styles easily?
+* Stability - can we safely change our styles without worrying about unrelated breakages?
+* Expressiveness - can we understand what our styles do, and do they make sense?
+* Performance - are our styles fast to load and render?
+
+If this seems overwhelming, don't worry, we're here to help.
```
Comment (craigjbass) [inline core-skills/frontend-web-development/resources/css-architecture.md]: I'm concerned that it's hard to refer back to how we solve these areas currently

File: core-skills/frontend-web-development/resources/css-architecture.md (hunk 1 above)
Comment (kjdchapman) [inline core-skills/frontend-web-development/resources/css-architecture.md]: this looks like a placeholder

Comment (craigjbass) [inline core-skills/frontend-web-development/resources/css-architecture.md]:
```suggestion
On the one hand, CSS Frameworks are usually battle-tested in multiple browsers. Leveraging one can provide the benefit of support in many browsers. It is also much faster to get going with a CSS Framework, as many common components are likely implemented for you.

The downside is that many CSS frameworks are usually very large slabs of CSS code, which will have a significant impact on the size of your CSS assets - larger assets mean that your page load times will become slower.

Another criticism is that CSS frameworks allow you to do things their way, but provide little-to-no help when you want to implement a custom set of styles, and can sometimes hinder this process.
```

Comment (craigjbass) [reply core-skills/frontend-web-development/resources/css-architecture.md]: ^ @SebAshton something like this?

**Pull request merged**